docs/contributors/code/testing-overview.md

```diff
@@ -19,9 +19,9 @@ When writing tests consider the following:
 
 ## JavaScript Testing
 
-Tests for JavaScript use [Jest](https://jestjs.io/) as the test runner and its API for [globals](https://jestjs.io/docs/en/api.html) (`describe`, `test`, `beforeEach` and so on) [assertions](https://jestjs.io/docs/en/expect.html), [mocks](https://jestjs.io/docs/en/mock-functions.html), [spies](https://jestjs.io/docs/en/jest-object.html#jestspyonobject-methodname) and [mock functions](https://jestjs.io/docs/en/mock-function-api.html). If needed, you can also use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) for React component testing.
+Tests for JavaScript use [**Jest**](https://jestjs.io/) as the test runner and its API for [globals](https://jestjs.io/docs/en/api.html) (`describe`, `test`, `beforeEach` and so on) [assertions](https://jestjs.io/docs/en/expect.html), [mocks](https://jestjs.io/docs/en/mock-functions.html), [spies](https://jestjs.io/docs/en/jest-object.html#jestspyonobject-methodname) and [mock functions](https://jestjs.io/docs/en/mock-function-api.html). Tests for React components should use [**React Testing Library**](https://testing-library.com/docs/react-testing-library/intro).
```

Comment:
Is this change necessary? Is there anything wrong with using React testing tools directly?

Reply:
@gziolo Do you mean `react-test-renderer` and `react-dom/test-utils`? I would highly recommend against `react-test-renderer`, which is worse than Enzyme. `react-dom/test-utils` could work for simple scenarios, but it's ultra verbose compared to RTL. You can see a comparison between different testing approaches here: https://github.com/WordPress/gutenberg/issues/17249#issuecomment-544337655

Reply:
React still keeps both those libraries/utils documented: 
https://reactjs.org/docs/test-utils.html

At the same time, they recommend RTL which isn't as strong as the phrase "should use". In addition to that, `act` from RTL is a simple wrapper for `react-dom/test-utils`. So there is still value in those primitives that React team provided. My main concern is that enforcing one library will impact the review process where the quality of tests would become less important that it's internal implementation.

Reply:
Those are all fair points!

My bad, I initially thought that we'd want to push for RTL with a bit more than a recommendation. I will undo the changes and leave that paragraph untouched.

Reply:
We can add more recommendations to help people make the right choice which we all agree that RTL is. We can also add an ESLint rule that errors when someone adds `enzyme` in the test file (we would have to add exceptions in existing test files though).

Reply:
Done in [`5dc8719` (#34423)](https://github.com/WordPress/gutenberg/pull/34423/commits/5dc8719f9ecf1d31e7515393d50dc50f9fed2cfd)

Reply:
> We can also add an ESLint rule that errors when someone adds enzyme in the test file (we would have to add exceptions in existing test files though).

Makes sense, I'll try to fit it in in the next days

Reply:
Tracked in #34437